docs/content/en/contribute/docs/source-file-structure/_index.md

```diff
@@ -54,6 +54,118 @@ In the order they appear in the rendered docs, the subdirectories are:
   Information to help users who are migrating to Keptn
   from Keptn v1
 
+### Working with reference pages
+
+The Keptn documentation includes two reference sections
+that document the Keptn APIs and CRDs.
+For background information, see:
+
+* [Kubernetes API Concepts](https://kubernetes.io/docs/reference/using-api/api-concepts/)
+* [Kubernetes API Reference](https://kubernetes.io/docs/reference/kubernetes-api/)
+
+#### API Reference
+
+The
+[API Reference](https://lifecycle.keptn.sh/docs/crd-ref/)
+pages are autogenerated from the source code.
+This is a comprehensive list of all APIs and resources Keptn uses,
+most of which users seldom see.
+
+Descriptive text for the APIs is authored in the source code itself.
+Each operator has its own API library with different versions.
+The source locations are:
+
+* [Lifecycle API](https://github.com/keptn/lifecycle-toolkit/tree/main/lifecycle-operator/apis/lifecycle)
+* [Metrics API](https://github.com/keptn/lifecycle-toolkit/tree/main/metrics-operator/api)
+* [Options API](https://github.com/keptn/lifecycle-toolkit/tree/main/lifecycle-operator/apis/options)
+
+The text is coded in a limited form of markdown.
+
+After modifying text for the API reference, you must regenerate it.
+
+To manually install the tool to regenerate the API reference docs,
+execute the following command
+in the root directory of your `lifecycle-toolkit` clone:
+
+```shell
+go install github.com/elastic/crd-ref-docs@master
+
+```
+
+To regenerate the autogenerated API reference docs,
+execute the following script
+in the root directory of your `lifecycle-toolkit` clone:
+
+```shell
+./.github/scripts/generate-crd-docs/generate-crd-docs.sh
+```
+
+### CRD Reference
+
+The
+[CRD Reference](https://lifecycle.keptn.sh/docs/yaml-crd-ref/)
+pages
+describe the YAML files used to populate resources
+for the small set of CRDs that users must populate themselves.
+These are currently authored manually
+and should provide as much information as possible about the resource.
+These are intended to be reference pages that are used regularly
+by users after they are familiar with how to use Keptn.
+For new users, the
+[User Guides](https://lifecycle.keptn.sh/docs/implementing/)
+provide introductory material about how to use various features.
+
+Here is a template to use to create a new CRD Reference page:
+
+```markdown
```

Comment:
I would put this template to a separate file and link it from here

Reply:
That sounds good but where would we put it?

Reply:
you can create a new `assets` folder next to the file and put it in there

Reply:
Is a name like yaml-crd-ref-template.md a good name for the file?

Reply:
sure, sounds good!

Reply:
Putting this file in assets didn't work out so well -- looks like it would work with some flavors of markdown but not all.  So I created a separate `templates` folder and put it in there.